test/humanStandardToken.js

```diff
@@ -1,345 +1,186 @@
-    }).then(function (result) {
-      var approvalLog = result.logs.find((element) => {
-        if (element.event.match('Approval')) { return true } else { return false }
-      })
-      assert.strictEqual(approvalLog.args._owner, accounts[0])
-      assert.strictEqual(approvalLog.args._spender, accounts[1])
-      assert.strictEqual(approvalLog.args._value.toString(), '2666')
-    }).catch((err) => { throw new Error(err) })
+  it('transfers: should transfer 10000 to accounts[1] with accounts[0] having 10000', async () => {
+    await HST.transfer(accounts[1], 10000, {from: accounts[0]})
+    const balance = await HST.balanceOf.call(accounts[1])
+    assert.strictEqual(balance.toNumber(), 10000)
+  })
+
+  it('transfers: should fail when trying to transfer 10001 to accounts[1] with accounts[0] having 10000', () => {
+    return expectThrow(HST.transfer.call(accounts[1], 10001, {from: accounts[0]}))
+  })
+
+  it('transfers: should handle zero-transfers normally', async () => {
+    assert(await HST.transfer.call(accounts[1], 0, {from: accounts[0]}), 'zero-transfer has failed')
+  })
+
+  // NOTE: testing uint256 wrapping is impossible in this standard token since you can't supply > 2^256 -1
+  // todo: transfer max amounts
+
+  // APPROVALS
+  it('approvals: msg.sender should approve 100 to accounts[1]', async () => {
+    await HST.approve(accounts[1], 100, {from: accounts[0]})
+    const allowance = await HST.allowance.call(accounts[0], accounts[1])
+    assert.strictEqual(allowance.toNumber(), 100)
+  })
+
+  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient. It should succeed.', async () => {
+    let SRS = await SampleRecipientSuccess.new({from: accounts[0]})
+    await HST.approveAndCall(SRS.address, 100, '0x42', {from: accounts[0]})
+    const allowance = await HST.allowance.call(accounts[0], SRS.address)
+    assert.strictEqual(allowance.toNumber(), 100)
+    const value = await SRS.value.call()
+    assert.strictEqual(value.toNumber(), 100)
+  })
+
+  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient and throw.', async () => {
+    let SRS = await SampleRecipientThrow.new({from: accounts[0]})
+    return expectThrow(HST.approveAndCall.call(SRS.address, 100, '0x42', {from: accounts[0]}))
+  })
+
+  // bit overkill. But is for testing a bug
+  it('approvals: msg.sender approves accounts[1] of 100 & withdraws 20 once.', async () => {
+    const balance0 = await HST.balanceOf.call(accounts[0])
+    assert.strictEqual(balance0.toNumber(), 10000)
+    await HST.approve(accounts[1], 100, {from: accounts[0]}) // 100
+    const balance2 = await HST.balanceOf.call(accounts[2])
+    assert.strictEqual(balance2.toNumber(), 0, 'balance2 not correct')
+    // transferFrom(address _from, address _to, uint256 _value)
+    HST.transferFrom.call(accounts[0], accounts[2], 20, {from: accounts[1]})
+    // allowance(address _owner, address _spender)
```

Comment:
What is the reason for this comment and the one above? I think they can/should be removed.

Reply:
As the tests are also used for documentation, I like to add function signature sometimes to avoid having to search in the code. I can remove it if you prefer.